src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1528,3 +1528,135 @@ def get_multisite_status(self):
+
+                metadata_sync_data['totalShards'] = metadata_sync_data['incrementalSync'][1]
+                metadata_sync_data['usedShards'] = int(metadata_sync_data['incrementalSync'][1]) - int(metadata_behind_shards)
+
+        else:
+            metadata_sync_data = metadata_sync_info
+
+        if 'data sync source' in metadata_sync_infoormation:
+            datasync_info = metadata_sync_infoormation.split('data sync source')[1].split('source:')
+
+        if datasync_info:
+            for index, item in enumerate(datasync_info):
+                replica_zone_data = {}
+                datasync_info_array = item.split('\n')
+
+                if len(datasync_info_array) > 0:
+                    replica_zone_name = self.get_primary_zonedata(datasync_info_array[0])
+                    sync_status = datasync_info_array[1]
+                    full_sync = ''
+                    incremental_sync = ''
+                    data_sync_status = ''
+
+                    for datasync_item in datasync_info_array:
+                        if 'full sync' in datasync_item and datasync_item.endswith('shards'):
+                            full_sync = datasync_item
+                        if 'incremental sync' in datasync_item:
+                            incremental_sync = datasync_item
+                        if 'data is behind' in datasync_item or 'data is caught up' in datasync_item:
+                            data_sync_status = datasync_item
+
+                    replica_zone_data['name'] = replica_zone_name.strip()
+                    replica_zone_data['fullSyncStatus'] = datasync_info_array
+                    replica_zone_data['syncstatus'] = sync_status.strip()
+                    replica_zone_data['fullSync'] = self.get_shards_info(full_sync.strip()).split('/') if full_sync.strip() != '' else []
+                    replica_zone_data['incrementalSync'] = self.get_shards_info(incremental_sync.strip()).split('/') if incremental_sync.strip() != '' else []
+                    replica_zone_data['dataSyncStatus'] = data_sync_status.strip()
+                    behind_shards = '0'
+                    if 'data is behind' in data_sync_status:
+                        behind_shards = self.get_behind_shards(data_sync_status)
+                    replica_zone_data['totalShards'] = int(replica_zone_data['incrementalSync'][1]) if len(replica_zone_data['incrementalSync']) > 1 else 0
+                    replica_zone_data['usedShards'] = (int(replica_zone_data['incrementalSync'][1]) - int(behind_shards)) if len(replica_zone_data['incrementalSync']) > 1 else 0
+                    replica_zones_info.append(replica_zone_data)
+
+        replica_zones_info_object['metadataSyncInfo'] = metadata_sync_data
+        replica_zones_info_object['dataSyncInfo'] = replica_zones_info
+        replica_zones_info_object['primaryZoneData'] = primary_zone_data
+        return replica_zones_info_object
```

Comment:
This is not clean and can't be reviewed. I see major concerns on this function and also can't decode how many thing are happening. 

1. Divide this function in multiple functional units. 
2. Give proper names to those functions and helper text on what all we are doing. 
3. Try to remove IF and use Objects to store and find data.
4. Try reducing the complexity of code to N or N(logn)
5. Make sure python readability is maintained. 
6. Also try to make sure function size needs is not bigger than 15 lines.

Reply:
Done. Thanks @cloudbehl